pkg/login/social/generic_oauth.go

```diff
@@ -260,11 +292,21 @@ func (s *SocialGenericOAuth) extractToken(data *UserInfoJson, token *oauth2.Toke
 	return true
 }
 
-func (s *SocialGenericOAuth) extractEmail(data *UserInfoJson) string {
+func (s *SocialGenericOAuth) extractEmail(userInfoResp []byte) string {
+	var data UserInfoJson
+	if err := json.Unmarshal(userInfoResp, &data); err != nil {
+		return ""
+	}
```

Comment:
If we pass the raw response as an additional parameter instead of replacing `data` we can avoid this additional unmarshaling.